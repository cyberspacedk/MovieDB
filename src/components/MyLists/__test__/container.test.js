import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import MyListsContainer from '../container';

describe('Container: MyListsContainer', () => {
  const store = configureStore()({
    myLists: {
      ids: [1, 2, 3],
      total_results: 10,
      current_page: 2,
      loading: false,
      error: false,
    },
    database: {
      lists: {
        1: { id: 1 },
      },
    },
  });

  const wrapper = shallow(<MyListsContainer store={store} />);
  const container = wrapper.dive();

  it('Snapshot: should match', () => {
    expect(container).toMatchSnapshot();
  });
});
