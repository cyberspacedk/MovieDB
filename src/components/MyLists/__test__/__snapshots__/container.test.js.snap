// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container: MyListsContainer Snapshot: should match 1`] = `
<MyList
  currentPage={2}
  deleteListRequest={[Function]}
  empty={false}
  error={false}
  getCreatedListRequest={[Function]}
  goToNextPage={[Function]}
  hideModal={[Function]}
  loading={false}
  myLists={
    Array [
      Object {
        "id": 1,
      },
      undefined,
      undefined,
    ]
  }
  showModal={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "payload": 1,
              "type": "GET_CREATED_LIST_REQUEST",
            },
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      },
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  totalResults={10}
  visibleMod={false}
/>
`;
